go-controller/pkg/node/vrfmanager/vrf_manager.go

```diff
@@ -20,20 +21,24 @@ import (
 type vrf struct {
 	name  string
 	table uint32
+	index int
 	// managedSlave is the desired netlink interface who's master will be this VRF.
 	// It cannot be changed after VRF creation.
 	managedSlave string
+	routes       []netlink.Route
 }
 
 type Controller struct {
-	mu   *sync.Mutex
-	vrfs map[int]vrf
+	mu           *sync.Mutex
+	vrfs         map[int]vrf
+	routeManager *routemanager.Controller
 }
 
-func NewController() *Controller {
+func NewController(routeManager *routemanager.Controller) *Controller {
```

Comment:
why couple these? why not let the consumer of this VRF controller to also manage the route controller? It can accomplish the same thing without coupling here.

Reply:
its easier to couple these because in the end its part of VRF Manager aspects... but more importantly I didn't want to have standalone routemanager inside UserDefinedNetworkGateway struct -> thought it would be cleaner to do it this way,,,
in fact when you, me and Peri met what feels like eons ago when I took over the PR we actually wanted to couple these together and not separate them which is why I did this change,,, (there should be a comment somewhere in his old PR) => if you have strong preferences we could do a FUP?